app/pulp/app/models/progress.py

```diff
@@ -18,6 +20,16 @@ class ProgressReport(Model):
 
     All progress reports have a message, state, and are related to a Task.
 
+    Attributes:
+
+        batch_increments (bool): If this Flag is set save checks to make sure it only gets ran once
+            every batch_interval milliseconds. This Flag defaults to False, it's set to true if the ProgressReport
+            is being used as a Context Manager.
+        last_increment_time (datetime.datetime): The last time save was called. This is used with
```

Comment:
I don't think the `last_increment_time` should be a public interface. A user should never set it. Can this become a private variable that is also undocumented.